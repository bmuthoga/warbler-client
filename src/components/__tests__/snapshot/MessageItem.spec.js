import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import MessageItem from '../../MessageItem'

const messageItemProps= {
  date: '2019-11-18T01:31:05.005Z',
  text: 'Sample message text',
  username: 'testemail',
  profileImageUrl: 'http://lorempixel.com/200/200/',
  removeMessage: jest.fn()
}

describe('MessageItem Snapshot Tests', () => {
  it('should render the MessageItem component without the Delete message button if the message belongs to another user', () => {
    const wrapper = shallow(
      <MessageItem { ...messageItemProps } />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render the MessageItem component with the Delete message button if the message belongs to the current user', () => {
    const wrapper = shallow(
      <MessageItem { ...messageItemProps } isCorrectUser={true} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
