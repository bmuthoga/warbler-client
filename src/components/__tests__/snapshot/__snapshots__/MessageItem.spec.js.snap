// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem Snapshot Tests should render the MessageItem component with the Delete message button if the message belongs to the current user 1`] = `
<div>
  <li
    className="list-group-item"
  >
    <img
      alt="testemail"
      className="timeline-image"
      height="100"
      src="https://lorempixel.com/200/200/"
      width="100"
    />
    <div
      className="message-area"
    >
      <Link
        to="/"
      >
        @
        testemail
      </Link>
       
      <span
        className="text-muted"
      >
        <t
          ago={false}
          calendar={false}
          className="text-muted"
          decimal={false}
          element={null}
          filter={[Function]}
          format="DD MMM YYYY"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-11-18T01:31:05.005Z
        </t>
      </span>
      <p>
        Sample message text
      </p>
      <a
        className="btn btn-danger"
        onClick={[MockFunction]}
      >
        Delete
      </a>
    </div>
  </li>
</div>
`;

exports[`MessageItem Snapshot Tests should render the MessageItem component without the Delete message button if the message belongs to another user 1`] = `
<div>
  <li
    className="list-group-item"
  >
    <img
      alt="testemail"
      className="timeline-image"
      height="100"
      src="https://lorempixel.com/200/200/"
      width="100"
    />
    <div
      className="message-area"
    >
      <Link
        to="/"
      >
        @
        testemail
      </Link>
       
      <span
        className="text-muted"
      >
        <t
          ago={false}
          calendar={false}
          className="text-muted"
          decimal={false}
          element={null}
          filter={[Function]}
          format="DD MMM YYYY"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-11-18T01:31:05.005Z
        </t>
      </span>
      <p>
        Sample message text
      </p>
    </div>
  </li>
</div>
`;
